helix-core/src/main/java/org/apache/helix/cloud/azure/AzureCloudInstanceInformation.java

```diff
@@ -0,0 +1,88 @@
+package org.apache.helix.cloud.azure;
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import org.apache.helix.api.cloud.CloudInstanceInformation;
+
+
+public class AzureCloudInstanceInformation implements CloudInstanceInformation {
+  private String _instanceName;
+  private String _faultDomain;
+  private String _instanceSetName;
+
+  /**
+   * Instantiate the AzureCloudInstanceInformation using each field individually.
+   * Users should use AzureCloudInstanceInformation.Builder to create information.
+   * @param instanceName
+   * @param faultDomain
+   * @param instanceSetName
+   */
+  public AzureCloudInstanceInformation(String instanceName, String faultDomain,
+      String instanceSetName) {
+    _instanceName = instanceName;
+    _faultDomain = faultDomain;
+    _instanceSetName = instanceSetName;
+  }
+
+  @Override
+  public String get(CloudInstanceField field) {
+    switch (field) {
```

Comment:
Let's use a map to store these information. Otherwise, in the future, if we need more fields, we need to add more property.